src/test/java/support/fakes/processors/StorageRecordPreProcessors.java

```diff
@@ -90,7 +94,40 @@ private StorageRecordPreProcessors() {
     });
   }
 
+  public static CompletableFuture<JsonObject> setStatusDateProcessor(
+    JsonObject oldItem, JsonObject newItem) {
+
+    // Create case
+    if (oldItem == null) {
+      return completedFuture(newItem);
+    }
+
+    return findHoldingForItem(newItem).thenApply(holding -> {
+      final String oldStatus = getNestedProperty(oldItem, "status", "name");
+      final String newStatus = getNestedProperty(newItem, "status", "name");
+
+      if (!Objects.equals(oldStatus, newStatus)) {
+        JsonObject newStatusObject = newItem.containsKey("status")
+          ? newItem.getJsonObject("status")
+          : new JsonObject();
+
+        newStatusObject = newStatusObject.put("date",
+          DateTime.now(DateTimeZone.UTC)
+            // RMB format
+            .toString("yyyy-MM-dd'T'HH:mm:ss.SSS+0000")
```

Comment:
Does RAML Module Builder always use this format?

Reply:
Jackson uses this pattern by default

Reply:
Does that mean that RAML Module Builder also does this by default?

Reply:
Yes, RMB uses Jackson for serialization/deserialization.

Reply:
Does it definitely use the default formatting?

If so, that is a shame, because that format isn't compliant with RFC-3339. 

I think we should use a format that is compliant.

Reply:
Here is `90000` item from snapshot:

![image](https://user-images.githubusercontent.com/9094650/71585322-cb6aaa80-2b1e-11ea-92c8-c8bae71d5701.png)

The `lastCheckIn.dateTime` and `metadata.updatedDate/createdDate` have format: date-time in json schema.

As you can see, they uses the `yyyy-MM-dd'T'HH:mm:ss.SSS+0000` format.

Reply:
Ok, that's really a bug. For the purposes of this change, can you extract a named constant with comments that explain this issue.

Reply:
Extracted, please take a look.